.github/workflows/storybook-chromatic-main.yml

```diff
@@ -1,13 +1,29 @@
 name: QA
 
 on:
-  push:
+  pull_request:
     branches:
-      - "main"
+      - main
+    types: [closed]
 
 jobs:
+  if_merged:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    outputs:
+      labels: ${{ steps.get_labels.outputs.labels }}
+    steps:
+      - id: get_labels
+        run: |
+          labels="$(gh api repos/${{ github.repository_owner }}/${{ github.event.repository.name }}/pulls/${{ github.event.pull_request.number }} --jq '.labels.[].name')"
+          echo "labels=$labels" >> "$GITHUB_OUTPUT"
```

Comment:
Follow security guidelines: https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#understanding-the-risk-of-script-injections